fleetspeak/src/e2etesting/lib/setup_components.go

```diff
@@ -0,0 +1,302 @@
+	if err != nil {
+		return fmt.Errorf("Unable to create textservices directory: %v", err)
+	}
+	err = os.Mkdir(path.Join(tempPath, "services"), 0777)
+	if err != nil {
+		return fmt.Errorf("Unable to create services directory: %v", err)
+	}
+	_, err = os.Create(path.Join(tempPath, "communicator.txt"))
+	if err != nil {
+		return fmt.Errorf("Unable to create communicator.txt: %v", err)
+	}
+	err = ioutil.WriteFile(path.Join(tempPath, "textservices", "frr.textproto"), []byte(proto.MarshalTextString(&clientServiceConf)), 0644)
+	if err != nil {
+		return fmt.Errorf("Unable to write frr.textproto file: %v", err)
+	}
+
+	// Server services configuration
+	serverServiceConf := servicesPb.ServiceConfig{Name: "FRR_server", Factory: "GRPC"}
+	grpcConfig := grpcServicePb.Config{Target: fmt.Sprintf("localhost:%v", fsFrontendPort), Insecure: true}
+	serverServiceConf.Config, err = ptypes.MarshalAny(&grpcConfig)
+	if err != nil {
+		return fmt.Errorf("Failed to marshal grpcConfig: %v", err)
+	}
+	serverConf := servicesPb.ServerConfig{Services: []*servicesPb.ServiceConfig{&serverServiceConf}, BroadcastPollTime: new(duration.Duration)}
+	serverConf.BroadcastPollTime.Seconds = 1
+	builtServerServicesConfigPath := path.Join(tempPath, "server.services.config")
+	err = ioutil.WriteFile(builtServerServicesConfigPath, []byte(proto.MarshalTextString(&serverConf)), 0644)
+	if err != nil {
+		return fmt.Errorf("Unable to write server.services.config: %v", err)
+	}
+	return nil
+}
+
+func (cc *componentCmds) start(tempPath string, fsFrontendPort, fsAdminPort int) error {
+	// Start server
+	cc.serverCmd = exec.Command("server", "-logtostderr", "-components_config", path.Join(tempPath, "server.config"), "-services_config", path.Join(tempPath, "server.services.config"))
+	go startProcess(cc.serverCmd)
+
+	serverStartTime := time.Now()
+
+	// Start client
+	cc.clientCmd = exec.Command("client", "-logtostderr", "-config", path.Join(tempPath, "linux_client.config"))
+	go startProcess(cc.clientCmd)
+
+	// Get new client's id and start service in current process
+	clientID, err := waitForNewClientID(fsAdminPort, serverStartTime)
+	if err != nil {
+		return fmt.Errorf("No new clients have connected: %v", err)
+	}
+	cc.serviceCmd = exec.Command(
+		"python",
+		"frr_python/frr_server.py",
+		fmt.Sprintf("--client_id=%v", clientID),
+		fmt.Sprintf("--fleetspeak_message_listen_address=localhost:%v", fsFrontendPort),
+		fmt.Sprintf("--fleetspeak_server=localhost:%v", fsAdminPort))
+	startProcess(cc.serviceCmd)
+	return nil
+}
+
+func run() error {
+	flag.Parse()
+
+	fsFrontendPort := 6062
+	fsAdminPort := 6061
+	tempPath, err := ioutil.TempDir(os.TempDir(), "*_fleetspeak")
+	if err != nil {
+		return fmt.Errorf("Failed to create temporary dir: %v", err)
+	}
+
+	err = configureFleetspeak(tempPath, fsFrontendPort, fsAdminPort)
+	if err != nil {
+		return fmt.Errorf("Failed to configure Fleetspeak: %v", err)
+	}
+
+	var cc componentCmds
+	defer cc.killAll()
```

Comment:
We  wait for the processes twice:

1) cc.killAll()
2) start calls startProcess() in go routine, which calls cmd.Wait()